velox/exec/tests/PlanNodeSerdeTest.cpp

```diff
@@ -623,4 +624,198 @@ TEST_F(PlanNodeSerdeTest, tableWriteWithStats) {
   testSerde(plan);
 }
 
+namespace {
+class MockIndexConnectorHandle : public connector::ConnectorTableHandle {
+ public:
+  explicit MockIndexConnectorHandle(std::string connectorId)
+      : ConnectorTableHandle(std::move(connectorId)) {}
+
+  ~MockIndexConnectorHandle() override = default;
+
+  std::string toString() const override {
+    static const std::string str{"MockIndexConnectorHandle"};
+    return str;
+  }
+
+  const std::string& name() const override {
+    static const std::string connectorName{"MockIndexConnectorHandle"};
+    return connectorName;
+  }
+
+  bool supportsIndexLookup() const override {
+    return true;
+  }
+
+  folly::dynamic serialize() const override {
+    folly::dynamic obj = folly::dynamic::object;
+    obj["name"] = name();
+    obj["connectorId"] = connectorId();
+    return obj;
+  }
+
+  static std::shared_ptr<MockIndexConnectorHandle> create(
+      const folly::dynamic& obj,
+      void* context) {
+    return std::make_shared<MockIndexConnectorHandle>(
+        obj["connectorId"].getString());
+  }
+
+  static void registerSerDe() {
+    auto& registry = DeserializationWithContextRegistryForSharedPtr();
+    registry.Register("MockIndexConnectorHandle", create);
+  }
+};
+
+class MockNonIndexConnectorHandle : public connector::ConnectorTableHandle {
+ public:
+  explicit MockNonIndexConnectorHandle(std::string connectorId)
+      : ConnectorTableHandle(std::move(connectorId)) {}
+
+  ~MockNonIndexConnectorHandle() override = default;
+
+  std::string toString() const override {
+    static const std::string str{"MockNonIndexConnectorHandle"};
+    return str;
+  }
+
+  const std::string& name() const override {
+    static const std::string connectorName{"MockNonIndexConnectorHandle"};
+    return connectorName;
+  }
+
+  bool supportsIndexLookup() const override {
+    return false;
+  }
+
+  folly::dynamic serialize() const override {
+    folly::dynamic obj = folly::dynamic::object;
+    obj["name"] = name();
+    obj["connectorId"] = connectorId();
+    return obj;
+  }
+
+  static std::shared_ptr<MockNonIndexConnectorHandle> create(
+      const folly::dynamic& obj,
+      void* context) {
+    return std::make_shared<MockNonIndexConnectorHandle>(
+        obj["connectorId"].getString());
+  }
+
+  static void registerSerDe() {
+    auto& registry = DeserializationWithContextRegistryForSharedPtr();
+    registry.Register("MockNonIndexConnectorHandle", create);
+  }
+};
+} // namespace
+
+TEST_F(PlanNodeSerdeTest, indexLookupJoin) {
+  MockNonIndexConnectorHandle::registerSerDe();
+  MockIndexConnectorHandle::registerSerDe();
+
+  auto mockIndexConnectorHandle =
```

Comment:
> mockIndexConnectorHandle

This name is rather long. It is a bit difficult to read and distinguish it from similarly looking mockNonIndexConnectorHandle. Consider renaming mockIndexConnectorHandle to indexHandle and using regular Hive connector for mockNonIndexConnectorHandle.